lib/src/error_matchers.dart

```diff
@@ -7,6 +7,9 @@ import 'type_matcher.dart';
 /// A matcher for [ArgumentError].
 const isArgumentError = TypeMatcher<ArgumentError>();
 
+/// A matcher for [AssertionError].
+const isAssertionError = TypeMatcher<AssertionError>();
```

Comment:
This will conflict with https://github.com/flutter/flutter/blob/95eed8764072bcf528b90a97a394fb2e07e4a026/packages/flutter_test/lib/src/matchers.dart#L201 so we will need to coordinate rollout.

There are also a few definitions internally.

Reply:
How do you want to coordinate roll out?

Reply:
We'd need to:
- Evaluate whether and how this is breaking for flutter users - IE would  they start getting ambiguous import warnings.
- Sync with someone from the flutter team and make sure they are on board with removing the definition from their code.
- Determine how we can roll this forward without breaking users. Flutter pins versions so it should be possible, but it might mean re-exporting this one from the file where they currently define it.

@kevmoo, @jacob314 - do you want to help with this?